lua/plenary/class.lua

```diff
@@ -0,0 +1,81 @@
+---@brief [[
+---classic
+---
+---Copyright (c) 2014, rxi
+---@brief ]]
+
+---@class Object
+local Object = {}
+Object.__index = Object
+
+---Does nothing.
+---You have to implement this yourself for extra functionality when initializing
+---@param self Object
+function Object:new()
```

Comment:
should we return something here by default? I'm not sure.

Or we should probably error, so you can't call this and expect something to happen.